fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/ProtocolSupport.java

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.metric.protocol;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * An object to describe which protocols a metric can support and to invoke the apply for that protocol.
+ *
+ * In the case where a stack of metrics might be delegated to, white or black listing allows the handler to indicate
+ * whether delegation should be allowed.
+ */
+public class ProtocolSupport {
+
+    /**
+     * A substitute for Boolean to indicate Yes, No or Uknown in the case where delegates might service a signal.
+     */
+    public enum Accepts {
+        TRUE,
+        REJECT,  // This Protocol is not supported and probably shouldn't be supported in metrics that close over it
+        MAYBE  // This protocol isn't directly supported by this metric
+    }
+
+    private final Collection<String> blacklist;
+    private final Map<String, Protocol> protocolMap;
+
+    /**
+     * Constructor.
+     *
+     * @param protocols A collection of protocols this handler will support.
+     */
+    public ProtocolSupport(
+            Collection<Protocol> protocols
+    ) {
+        this(protocols, Collections.emptySet());
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param protocols A collection of protocols this handler will support.
+     * @param blacklist  Protocols that this handler will neither support nor delegate.
+     */
+    public ProtocolSupport(
+            Collection<Protocol> protocols,
+            Collection<String> blacklist
+    ) {
+        protocolMap = protocols.stream().collect(Collectors.toMap(Protocol::getContractName, Function.identity()));
+        this.blacklist = blacklist;
+    }
+
+    /**
+     * Determine if this protocol handler or accepts this protocol.
+     *
+     * @param contractName The name of the protocol to test for.
+     *
+     * @return TRUE if this metric directly or indirectly supports this signal, FALSE if it refuses, MAYBE if it
+     * doesn't assert authority.
+     */
+    public Accepts accepts(String contractName) {
+        return protocolMap.containsKey(contractName) ? Accepts.TRUE :
```

Comment:
Shouldn't the blacklist take priority? Right now, if it's in the protocol map, and in the blacklist, we'll return Accepts.TRUE, where I would expect us to return Accepts.REJECT.